### ISSUE TITLE
identity_conversion false positive with `?` for loops in 1.35

### ISSUE THREAD

**Issue by sfackler** [rust-lang/rust-clippy#4133]
This may be a duplicate of #4081, but I can't easily check since clippy isn't in the current nightly.

```rust
fn foo() -> Result<Vec<u8>, ()> {
    Err(())
}

fn bar() -> Result<(), ()> {
    for _ in foo()? {}

    Ok(())
}
```

```
warning: identical conversion
 --> src/main.rs:6:14
  |
6 |     for _ in foo()? {}
  |              ^^^^^^ help: consider removing `foo()?()`: `foo()?`
  |
  = note: #[warn(clippy::identity_conversion)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_conversion

```

**Comment by hoodie** [comment]
ditto: seeing lots of those when I iterare over `Result<Vec<_>>` with `?`

**Comment by baumanj** [comment]
Seeing this in 1.34 as well

**Comment by flip1995** [comment]
Yes this is fixed in #4083 and will be available in the next nightly Clippy.

**Comment by flip1995** [comment]
I think that would be the right thing to do.

**Comment by oli-obk** [comment]
@flip1995 	that issue link (4083) is not pointing to a PR. I have been unable to find the PR that fixed it.

**Comment by flip1995** [comment]
Oh sorry, it was the #4082 PR

**Comment by oli-obk** [comment]
I just checked, all of that PR is in the current beta, so we're good

**Comment by wraithan** [comment]
Looks like this is broken on the clippy that comes with stable via `rustup component add clippy` on rust 1.35.0

```
[~/devel/rust/shader-playground] master(1)+* ± rustc --version && cargo --version && cargo clippy -- --version
rustc 1.35.0 (3c235d560 2019-05-20)
cargo 1.35.0 (6f3e9c367 2019-04-04)
clippy 0.0.212 (265318db 2019-05-17)
[~/devel/rust/shader-playground] master(1)+* ± cargo clippy
    Checking shader-playground v0.1.0 (/Users/wraithan/devel/rust/shader-playground)
error: identical conversion
   --> src/shader.rs:181:18
    |
181 |     for entry in std::fs::read_dir(dir)? {
    |                  ^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `std::fs::read_dir(dir)?()`: `std::fs::read_dir(dir)?`
    |
```
